smithy-aws-protocol-tests/model/awsJson1_0/defaults.smithy

```diff
@@ -0,0 +1,357 @@
+$version: "2.0"
+
+namespace aws.protocoltests.json10
+
+use aws.protocols#awsJson1_0
+use smithy.test#httpRequestTests
+use smithy.test#httpResponseTests
+
+apply OperationWithDefaults @httpRequestTests([
+    {
+        id: "AwsJson10ClientPopulatesDefaultValuesInRequestBodyIfMissingInParams",
+        documentation: "Client populates default values in request body if missing in params.",
+        protocol: awsJson1_0,
+        method: "POST",
+        bodyMediaType: "application/json",
+        uri: "/",
+        headers: {
+            "Content-Type": "application/x-amz-json-1.0"
+        },
+        body: """
+            {
+                "defaultString": "hi",
+                "defaultBoolean": true,
+                "defaultList": [],
+                "defaultDocumentMap": {},
+                "defaultDocumentString": "hi",
+                "defaultDocumentBoolean": true,
+                "defaultDocumentList": [],
+                "defaultNullDocument": null,
+                "defaultTimestamp": 0,
+                "defaultBlob": "{}",
+                "defaultByte": 1,
+                "defaultShort": 1,
+                "defaultInteger": 10,
+                "defaultLong": 100,
+                "defaultFloat": 0,
+                "defaultDouble": 0,
+                "defaultMap": {},
+                "defaultBigInteger": 0,
+                "defaultBigDecimal": 0,
+                "defaultEnum": "FOO",
+                "defaultIntEnum": 1
+            }""",
+        params: {
+        },
+        tags: ["defaults"],
```

Comment:
It would be great to move tags and appliesTo before long params blocks so it's easier to read these and know what they apply to.